src/format/formatting.rs

```diff
@@ -422,6 +362,75 @@ fn format_inner(
     }
 }
 
+#[cfg(feature = "alloc")]
+fn format_numeric(
```

Comment:
How about attaching this to `Numeric` as a method?

Reply:
Later commits make it a method on `DelayedFormat`. That also seems like a good place?

Reply:
Right, that seems to be actually better.